client/my-sites/stats/stats-post-anniversaries/index.jsx

```diff
@@ -0,0 +1,209 @@
+ */
+import ErrorPanel from '../stats-error';
+import StatsList from '../stats-list';
+import StatsListLegend from '../stats-list/legend';
+import SectionHeader from 'components/section-header';
+import StatsContentText from '../stats-module/content-text';
+import StatsModulePlaceholder from '../stats-module/placeholder';
+import Card from 'components/card';
+import QueryPosts from 'components/data/query-posts';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSitePostsForQuery, isRequestingSitePostsForQuery } from 'state/posts/selectors';
+
+// Maximum number of simultaneous queries to fetch the posts by year
+const MAX_POST_QUERIES = 10;
+
+const StatModulePostAnniversaries = props => {
+	const { oldestPostQuery, postsByYearQueries, postsByYear, requesting, siteId, translate } = props;
+
+	const allPosts = Array.prototype.concat( ...postsByYear );
+
+	const cardClasses = classNames( 'stats-module', 'is-expanded', 'summary', {
+		'is-loading': requesting,
+		'has-no-data': ! allPosts.length,
+		'is-showing-error': ! allPosts.length,
+	} );
+
+	return (
+		<div>
+			<SectionHeader
+				label={ translate( 'Anniversaries', {
+					context: 'Stats: title of module',
+				} ) }
+			/>
+			<Card className={ cardClasses }>
+				{ siteId && <QueryPosts siteId={ siteId } query={ oldestPostQuery } /> }
+				{ siteId &&
+					postsByYearQueries.map(
+						( query, i ) => <QueryPosts key={ i } siteId={ siteId } query={ query } />,
+					) }
+
+				<StatsModulePlaceholder isLoading={ requesting } />
+
+				{ ! requesting &&
+					allPosts &&
+					( allPosts.length === 0
+						? <EmptyMessage />
+						: <PostsList allPosts={ allPosts } postsByYear={ postsByYear } /> ) }
+			</Card>
+		</div>
+	);
+};
+
+const CalendarIcon = () => (
+	<span style={ { position: 'relative', top: '-2px' } }>
+		<Gridicon icon="calendar" size={ 18 } />
+	</span>
+);
+
+const PostsList = ( { allPosts, postsByYear } ) =>
+	( allPosts.length === 1
+		? <SinglePost post={ allPosts[ 0 ] } />
+		: <GroupedPosts postsByYear={ postsByYear } /> );
+
+const SinglePost = localize( ( { translate, post } ) => {
+	const yearsAgo = moment()
+		.startOf( 'year' )
+		.diff( moment( post.date ).startOf( 'year' ), 'years' );
+	return (
+		<StatsContentText>
+			<p>
+				<CalendarIcon />
+				{ translate(
+					'%(yearsAgo)d year ago on this day, {{href}}%(title)s{{/href}} was published.',
+					'%(yearsAgo)d years ago on this day, {{href}}%(title)s{{/href}} was published.',
+					{
+						count: yearsAgo,
+						args: { yearsAgo, title: post.title },
+						components: {
+							href: <a href={ post.URL } target="_blank" rel="noopener noreferrer" />,
+						},
+						context: 'Stats: Sentence showing what post was published some years ago',
+					},
+				) }
+			</p>
+		</StatsContentText>
+	);
+} );
+
+const GroupedPosts = localize( ( { translate, postsByYear } ) => (
+	<div>
+		<StatsContentText>
+			<p>
+				{ translate( 'Published on this day in the past years:', {
```

Comment:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?